test-suite/src/test/scala/org/scalajs/testsuite/javalib/ConcurrentSkipListSetTest.scala

```diff
@@ -0,0 +1,431 @@
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013-2015, LAMP/EPFL   **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+package org.scalajs.testsuite.javalib
+
+import scala.language.implicitConversions
+
+import scala.collection.JavaConversions._
+import scala.collection.mutable
+
+import org.scalajs.jasminetest.JasmineTest
+
+import scala.scalajs.runtime.UndefinedBehaviorError
+
+import java.util.concurrent.ConcurrentSkipListSet
+import java.util.Comparator
+
+object ConcurrentSkipListSetTest extends JasmineTest {
+
+  describe("java.util.concurrent.ConcurrentSkipListSet") {
+
+    it("should store and remove ordered integers") {
+      val csls = new ConcurrentSkipListSet[Int]()
+      
+      expect(csls.size()).toEqual(0)
+      expect(csls.add(222)).toBeTruthy
+      expect(csls.size()).toEqual(1)
+      expect(csls.add(111)).toBeTruthy
+      expect(csls.size()).toEqual(2)
+      expect(csls.first).toEqual(111)
+      expect(csls.remove(111)).toBeTruthy
+      
+      expect(csls.size()).toEqual(1)
+      expect(csls.first).toEqual(222)
+
+      expect(csls.remove(222)).toBeTruthy
+      expect(csls.size()).toEqual(0)
+      expect(csls.isEmpty).toBeTruthy
+      expect(csls.remove(333)).toBeFalsy
+      expect{
+          try {
+            csls.first
+            false
+          } catch {
+            case _ : NoSuchElementException => true
+            case _ : Throwable => false
+          }
+      }.toBeTruthy
+    }
+
+    it("should store and remove ordered strings") {
+      val csls = new ConcurrentSkipListSet[String]()
+      
+      expect(csls.size()).toEqual(0)
+      expect(csls.add("222")).toBeTruthy
+      expect(csls.size()).toEqual(1)
+      expect(csls.add("111")).toBeTruthy
+      expect(csls.size()).toEqual(2)
+      expect(csls.first).toEqual("111")
+      expect(csls.remove("111")).toBeTruthy
+      
+      expect(csls.size()).toEqual(1)
+      expect(csls.first).toEqual("222")
+
+      expect(csls.remove("222")).toBeTruthy
+      expect(csls.size()).toEqual(0)
+      expect(csls.remove("333")).toBeFalsy
+      expect(csls.isEmpty).toBeTruthy
+    }
+
+    when("compliant-asinstanceofs").
+    it("should throw exception on non comparable objects") {
+      case class TestObj(num: Int) 
+
+      val csls = new ConcurrentSkipListSet[TestObj]()
+      
+      expect(csls.size()).toEqual(0)
+      expect(csls.add(TestObj(222))).toBeTruthy
+      expect(csls.size()).toEqual(1)
+      expect{
+        try {
+          csls.add(TestObj(111))
+          false
+        } catch {
+          case err : UndefinedBehaviorError =>
+            err.getCause match {
+              case _ : ClassCastException => true
+              case _ => false
```

Comment:
When `compliant-asinstanceofs`, the exception that is thrown is a `ClassCastException`, not an `UndefinedBehaviorError` wrapping a `ClassCastException`. So you should catch directly a `ClassCastException` here.

To test this on your side:

    > set scalaJSSemantics in testSuite ~= { _.withAsInstanceOfs(org.scalajs.core.tools.sem.CheckedBehavior.Compliant }
    > testSuite/test